Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -154,6 +154,27 @@ public extension Sequence {
         return singleElement
     }
 
+    /// SwifterSwift: Remove duplicate elements based on condition.
+    ///
+    ///        [1, 2, 1, 3, 2].withoutDuplicates { $0 } -> [1, 2, 3]
+    ///        [(1, 4), (2, 2), (1, 3), (3, 2), (2, 1)].withoutDuplicates { $0.0 } -> [(1, 4), (2, 2), (3, 2)]
+    ///
+    /// - Parameter condition: conditions to evaluate repeating elements.
+    /// - Returns: Sequence without repeating elements
+    /// - Complexity: O(*n*), where *n* is the length of the sequence.
+    func withoutDuplicates<E: Hashable>(_ condition: (Element) -> E ) -> [Element] {
+        var set = Set<E>()
+        return self.filter {
```

Comment:
This can all go in 1 line.
```suggestion
        return filter { !set.insert(condition($0)).inserted }
```